src/runtime/crt/common/ndarray.c

```diff
@@ -76,7 +76,7 @@ int TVMNDArray_Load(TVMNDArray* ret, const char** strm) {
   *strm += sizeof(ndim);
   dtype = ((DLDataType*)*strm)[0];  // NOLINT(*)
   *strm += sizeof(dtype);
-  if ((ndim < 0) || (ndim > TVM_CRT_MAX_NDIM)) {
+  if (ndim > TVM_CRT_MAX_NDIM) {
```

Comment:
@wrongtest I think it would make more sense here to change the type of `ndim` to int to match the dlpack.h header:
https://github.com/dmlc/dlpack/blob/master/include/dlpack/dlpack.h#L136

then we'd need to keep the original check. would you be up for making this change?

Reply:
Sorry for not have noticed your comment later. I update related types of ndarray.c to int32_t. While original dlpack ndim is just defined as `int`, line 71 use also `int` instead of int32_t. 

I run a local compile with `-Wsign-compare -Wsign-conversion` to ensure not inconsistency introduced internally in this file.